src/main/java/lotto/model/LottoMachine.java

```diff
@@ -0,0 +1,60 @@
+package lotto.model;
+
+import lotto.utils.LottoCompare;
+
+import java.util.*;
+
+public class LottoMachine {
+    private final Lottos lottos;
+    private final WinningLotto winningLotto;
+    private final List<LottoCompare> lottoCompares;
+
+    public LottoMachine(Lottos lottos, WinningLotto winningLotto, int count) {
+        this.lottos = lottos;
+        this.winningLotto = winningLotto;
+        this.lottoCompares = initCompares(count);
+    }
+
+    private List<LottoCompare> initCompares(int count) {
+        List<LottoCompare> lottoCompareResults = new ArrayList<>();
+
+        for(int i = 0; i < count; i++) {
+            int match = compareLotto(lottos.getLottos().get(i));
+            boolean isIncludeBonusNumbers = compareBonus(lottos.getLottos().get(i));
+            LottoCompare lottoResult = LottoCompare.getCompareResult(match, isIncludeBonusNumbers);
+            lottoCompareResults.add(lottoResult);
+        }
```

Comment:
스트림을 사용한다면 더 간결해 질 수 있을 것 같습니다!
```suggestion
        lottos.getLottos().stream()
                    //LottoCompare.getCompareResult가 Lotto 객체를 
                    //파라미터로 받도록 리팩토링 한다면
                .map(LottoCompare::getCompareResult) 
                .forEach(lottoCompareResults::add);
```